pkg/cmd/cli/cmd/edit.go

```diff
@@ -306,6 +339,37 @@ func (r *editResults) AddError(err error, info *resource.Info) string {
 	}
 }
 
+type patchError struct {
+	error
+}
+
+// tryPatch reads the latest version of the object, writes it to version, then attempts to merge
+// the changes onto it without conflict. If a conflict occurs jsonmerge.IsConflicting(err) is
+// true. The info object is mutated
+func tryPatch(delta *jsonmerge.Delta, info *resource.Info, version string) error {
```

Comment:
I'm not a fan of direct mutation.  How would feel about returning a new `*resource.Info`?